cosmos/interchain-security
# fix e2e

- change ibc module paths
- Squashed commit of the following:
- Squashed commit of the following:
- remove buf.work.yaml and don't tidy when building protos
- rebuild proto files
- simapp, cometbft-db and cometbft
- don't use rest and update ibc client types
- tidy is successful
- resolve imports in app,go
- fix api change
- apply change in gov and staking in 47
- init genesis in democracy module
- fixing build
- update code for 47
- Squashed commit of the following:
- change govtypes, update ccvstaking, ccvgov, ccvdistr newAppModule
- modufy check whitelist content
- update consumer and provider keeper hooks
- update proof path
- update Slash in staking keeper's signature
- regenerate mocks for staking keeper
- tidy
- update provider hooks: return errors
- lint ccv/provider/keeper: sdkerrors + math.Int
- ccv/consumer/keeper: pass correct args to Slash()
- ccv/consumer/keeper: lint: migrate to cosmossdk.io/(errors/math)
- sdk.Int -> math.Int
- app: register newer GRPC services (https://github.com/cosmos/cosmos-sdk/blob/release/v0.47.x/UPGRADING.md#app-wiring)
- consumer-democracy: new signature for NewApp() + remove invCheckPeriod from App struct
- consumer: new signature for NewApp() + remove invCheckPeriod from App struct
- provider: new signature for NewApp() + remove invCheckPeriod from App struct
- Revert "ccv/consumer/keeper: pass correct args to Slash()"
- Revert "update Slash in staking keeper's signature"
- Revert "regenerate mocks for staking keeper"
- minor change
- minor change
- replace deprecated ibctranferkeeper func sendtransfer
- x/ccv/types: sdk.Int -> math.Int, update sdkerrors path
- regenerate mocks
- minor update mock
- remove all usage of rest
- slash functoin fix by fork sdk
- remove unnecessary SA1019 disabler
- global midration from sdkerrors to cosmossdk.io/errors
- regenerate mocks: exclude transfertypes import
- simtypes.WeightedProposalContent -> WeightedProposalMsg. Deprecated
- go fmt file
- update getParams for consumerkeeper and add SlashWithInfractionReason
- Update Makefile
- add unit tests to Makefile
- dockerignore
- Update .dockerignore
- change back to deleteProposal
- add consensus keeper to Apps + remove redundant invCheckPeriod
- rename ec.Marshaler to ec.Codec
- change all appearences of -b block to -b sync
- stragglers: Marshaler -> Codec migration
- lint docs
- modulesToExport: add to ExportAppStateAndValidators
- sdk.Int -> math.Int
- minor change
- remove test
- bump versions
- seperate two types of WhitelistedParams
- fix deleteForbiddenProposal
- revert legacy whitelist params and fix democracy gov whitelist test
- update forbidden ante test
- add comment and minor change
- clean up
- fix lints
- Create go-tests.yml
- run unit tests separately and use the verbose flag for e2e
- update LegacyWhitelistedParams: only whitelist ibc params
- trigger ci
- disable legacy parameter changes
- Revert "trigger ci"
- remove unnecessary initgenesis
- fix: unboding do not found
- root.go: cd and cdd don't need valset in genesis
- fix: use sdk47 staking keeper interface
- fix: change encoding used by provider app [unstable]
- fix: use mainline ibc-go/v7 in integration tests
- fix: refactor difftest setup procedure
- fix lints
- use a different version of go in docker
- update proto builder to latest
- update go.mod
- tidy
- apply MSalopek patch to cosmos-sdk so work may continue
- organize go.mod and add thanks
- bump the version of hermes used in docs and images
- run e2e verbose so we can see where it breaks
- cleanup root.go
- fix consumer democracy, begin fixes on consumer and producer
- tidy
- cleanup cd
- cleanup producer
- fix provider app
- use genesis command with add-genesis-account subcommand
- use genesis subcommand
- return only rootCmd
- add defaultMsgValidator to provider genutil
- trigger ci
- submit-legacy-proposal
- add err handler
- test is description summary or nah?
- Revert "test is description summary or nah?"
- dunno why this would work tbh...
- Revert "dunno why this would work tbh..."
- begin changing prop types
- summary
- summary
- use legacy proposal type with summary field
- update example proposal commands
- add type
- proposal type
- fix gov cli
- legacy router
- minor
- remove type flag in gov proposal, it's not exist
- async broadcast
- change broadcast type to sync
- add sleep time
- gov proposal changed
- gjson
- minor



## Description

Closes: #XXXX



---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...
